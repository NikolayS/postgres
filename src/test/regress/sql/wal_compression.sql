--
-- Test WAL compression functionality with method:level syntax
--

-- Test basic parameter existence and default value
SHOW wal_compression;

-- Test setting compression methods without levels
SET wal_compression = 'off';
SHOW wal_compression;

SET wal_compression = 'pglz';
SHOW wal_compression;

SET wal_compression = 'lz4';
SHOW wal_compression;

SET wal_compression = 'zstd';
SHOW wal_compression;

-- Test setting compression methods with levels
SET wal_compression = 'lz4:1';
SHOW wal_compression;

SET wal_compression = 'lz4:9';
SHOW wal_compression;

-- Note: LZ4HC levels (10-12) are not tested here because their availability
-- depends on build configuration (HAVE_LZ4HC_H). On builds without LZ4HC,
-- these levels are rejected at SET time with appropriate error messages.

SET wal_compression = 'zstd:1';
SHOW wal_compression;

SET wal_compression = 'zstd:9';
SHOW wal_compression;

SET wal_compression = 'zstd:22';
SHOW wal_compression;

-- Test invalid compression levels (should fail)
SET wal_compression = 'lz4:0';
SET wal_compression = 'lz4:13';
SET wal_compression = 'zstd:0';
SET wal_compression = 'zstd:23';

-- Test PGLZ with levels (should fail)
SET wal_compression = 'pglz:1';

-- Test invalid compression methods (should fail)
SET wal_compression = 'invalid';
SET wal_compression = 'gzip';

-- Test malformed syntax (should fail)
SET wal_compression = 'lz4:';
SET wal_compression = 'lz4:abc';
SET wal_compression = ':5';

-- Reset to default
SET wal_compression = 'off';
SHOW wal_compression;
