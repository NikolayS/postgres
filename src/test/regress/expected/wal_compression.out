--
-- Test WAL compression functionality with method:level syntax
--
-- Test basic parameter existence and default value
SHOW wal_compression;
 wal_compression 
-----------------
 off
(1 row)

-- Test setting compression methods without levels
SET wal_compression = 'off';
SHOW wal_compression;
 wal_compression 
-----------------
 off
(1 row)

SET wal_compression = 'pglz';
SHOW wal_compression;
 wal_compression 
-----------------
 pglz
(1 row)

SET wal_compression = 'lz4';
SHOW wal_compression;
 wal_compression 
-----------------
 lz4
(1 row)

SET wal_compression = 'zstd';
SHOW wal_compression;
 wal_compression 
-----------------
 zstd
(1 row)

-- Test setting compression methods with levels
SET wal_compression = 'lz4:1';
SHOW wal_compression;
 wal_compression 
-----------------
 lz4:1
(1 row)

SET wal_compression = 'lz4:9';
SHOW wal_compression;
 wal_compression 
-----------------
 lz4:9
(1 row)

-- Note: LZ4HC levels (10-12) are not tested here because their availability
-- depends on build configuration (HAVE_LZ4HC_H). On builds without LZ4HC,
-- these levels are rejected at SET time with appropriate error messages.
SET wal_compression = 'zstd:1';
SHOW wal_compression;
 wal_compression 
-----------------
 zstd:1
(1 row)

SET wal_compression = 'zstd:9';
SHOW wal_compression;
 wal_compression 
-----------------
 zstd:9
(1 row)

SET wal_compression = 'zstd:22';
SHOW wal_compression;
 wal_compression 
-----------------
 zstd:22
(1 row)

-- Test invalid compression levels (should fail)
SET wal_compression = 'lz4:0';
ERROR:  invalid value for parameter "wal_compression": "lz4:0"
DETAIL:  LZ4 compression level must be between 1 and 12.
SET wal_compression = 'lz4:13';
ERROR:  invalid value for parameter "wal_compression": "lz4:13"
DETAIL:  LZ4 compression level must be between 1 and 12.
SET wal_compression = 'zstd:0';
ERROR:  invalid value for parameter "wal_compression": "zstd:0"
DETAIL:  ZSTD compression level must be between 1 and 22.
SET wal_compression = 'zstd:23';
ERROR:  invalid value for parameter "wal_compression": "zstd:23"
DETAIL:  ZSTD compression level must be between 1 and 22.
-- Test PGLZ with levels (should fail)
SET wal_compression = 'pglz:1';
ERROR:  invalid value for parameter "wal_compression": "pglz:1"
DETAIL:  Compression method "pglz" does not support compression levels.
-- Test invalid compression methods (should fail)
SET wal_compression = 'invalid';
ERROR:  invalid value for parameter "wal_compression": "invalid"
DETAIL:  Unrecognized compression method: "invalid".
SET wal_compression = 'gzip';
ERROR:  invalid value for parameter "wal_compression": "gzip"
DETAIL:  Unrecognized compression method: "gzip".
-- Test malformed syntax (should fail)
SET wal_compression = 'lz4:';
ERROR:  invalid value for parameter "wal_compression": "lz4:"
DETAIL:  Unrecognized compression method: "lz4:".
SET wal_compression = 'lz4:abc';
ERROR:  invalid value for parameter "wal_compression": "lz4:abc"
DETAIL:  Unrecognized compression method: "lz4:abc".
SET wal_compression = ':5';
ERROR:  invalid value for parameter "wal_compression": ":5"
DETAIL:  Unrecognized compression method: ":5".
-- Reset to default
SET wal_compression = 'off';
SHOW wal_compression;
 wal_compression 
-----------------
 off
(1 row)

